src/worker/worker.ts

```diff
@@ -130,6 +133,21 @@ function applyUpdates({
   setStatus("idle")
 }
 
+function applyOptionsUpdate(newOptions: UpdatableArchipelagoParameters) {
```

Comment:
This seems quite repeated with applyUpdates. Maybe we can extract a method with a function as parameter? Something like `performArchipelagoOperation(operation: Archipelago => IslandUpdates)`